Source/DafnyCore/ErrorDetail.cs

```diff
@@ -10,14 +10,61 @@ public static class ErrorDetail {
   public enum ErrorID {
```

Comment:
Why do we need this enum? How about using strings for the error codes wherever we create the error message? A very large enum with a value for each error message doesn't seem right to me. How does this scale to plugins with error messages in different assemblies that cannot be reached from `DafnyCore` ?

Reply:
In early discussion with Mikhael on this design we debated among:
1) using just constant string literals everywhere
2) using 'const string abc = "abc";' declarations and then using the name everywhere
3) using an enum

The first seemed too prone to easily missed misspellings
The second seemed verbose.
The third seemed best.

Reply:
I wanted to use the enum just to avoid repeating a variable name and a constant with the same value as the variable name.
If you have a way to do that @keyboardDrummer so that it's compatible with any other user-provided strings, then I think the would be the best